platform/o.n.core/test/qa-functional/src/org/netbeans/core/validation/ValidateClassFilesTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.core.validation;
+
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Enumeration;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import junit.framework.Test;
+import org.netbeans.junit.NbModuleSuite;
+import org.netbeans.junit.NbTestCase;
+
+public class ValidateClassFilesTest extends NbTestCase {
+    public ValidateClassFilesTest(String name) {
+        super(name);
+    }
+
+    public static Test suite() {
+        return NbModuleSuite.createConfiguration(ValidateClassFilesTest.class).
+                clusters(".*").gui(false).honorAutoloadEager(true).suite();
+    }
+
+    public void testCheckClassFiles() throws IOException {
```

Comment:
Yes, the goal is to ensure class version is at most 52 (Java 8) unless otherwise stated. Right now there is an exception for `extra/modules/*javafx*.jar` which are only available/used on JDK11.